### PR TITLE
Disable/hide CRUD buttons if org mismatch

### PR DESCRIPTION
**Pull request by katybaulch**
# What's changed

[PDCT-1104](https://linear.app/climate-policy-radar/issue/PDCT-1104/block-frontend-crud-if-org-mismatch) 

Block CRUD when org mismatch.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

### CODE REVIEW

File: src/components/forms/DocumentForm.tsx [hunk 1]
```diff
@@ -265,7 +267,7 @@ export const DocumentForm = ({
               )
             }}
           />
-          <ButtonGroup>
+          <ButtonGroup isDisabled={canAccess}>
```
Comment (diversemix) [inline src/components/forms/DocumentForm.tsx:270]: Is this logic right? I would have thought `isDisabled= !canAccess`

**Review by diversemix — APPROVED** [review]
Be worth checking that logic!

File: src/components/buttons/Delete.tsx [hunk 2]
```diff
@@ -62,9 +68,28 @@ export const DeleteButton = ({ entityName, entityTitle, callback }: IProps) => {
             <Button ref={cancelRef} onClick={onClose}>
               No
             </Button>
-            <Button colorScheme='red' ml={3} onClick={handleConfirmClick}>
-              Yes
-            </Button>
+            {!isDisabled && (
```
Comment (PatFawbertMills) [inline src/components/buttons/Delete.tsx:71]: I think we can do this in a more DRY way - let's jump on a call

File: src/components/forms/DocumentForm.tsx (hunk 1 above)
Comment (PatFawbertMills) [inline src/components/forms/DocumentForm.tsx:270]: Should this not be `!canAccess`?

File: src/components/forms/EventForm.tsx [hunk 3]
```diff
@@ -168,7 +170,7 @@ export const EventForm = ({
               )
             }}
           />
-          <ButtonGroup>
+          <ButtonGroup isDisabled={canAccess}>
```
Comment (PatFawbertMills) [inline src/components/forms/EventForm.tsx:173]: As above

**Review by PatFawbertMills — CHANGES_REQUESTED** [review]

**Review by PatFawbertMills — CHANGES_REQUESTED** [review]
Could you confirm whether a user should be able to delete a family of a different org?

At the moment this is still possible from the document list

File: src/components/buttons/Delete.tsx (hunk 2 above)
Comment (katybaulch) [reply src/components/buttons/Delete.tsx]: Have pushed what we added :)

**Review by PatFawbertMills — APPROVED** [review]
Nice work and good catch on the locale date issue.

File: src/components/forms/EventForm.tsx (hunk 3 above)
Comment (katybaulch) [reply src/components/forms/EventForm.tsx]: Have resolved.

File: src/components/forms/DocumentForm.tsx (hunk 1 above)
Comment (katybaulch) [reply src/components/forms/DocumentForm.tsx]: Have resolved.

Comment (katybaulch) [reply src/components/forms/DocumentForm.tsx]: Have resolved.

**Pull request merged**